applications/FSIapplication/custom_utilities/mvqn_convergence_accelerator.hpp

```diff
@@ -373,96 +344,126 @@ class MVQNFullJacobianConvergenceAccelerator: public ConvergenceAccelerator<TSpa
     MatrixPointerType mpObsMatrixW;             // Solution increment observation matrix
 
     ///@}
-
-    ///@name Protected Operators
+    ///@name Member Variables
     ///@{
-    ///@}
 
-    ///@name Protected Operations
+    ///@}
+    ///@name Private Operators
     ///@{
 
-    template<class T>
-    bool InvertMatrix(const T& input, T& inverse)
+    /**
+     * @brief Append the new data to the observation matrices
+     * This function appends the new data to both observation matrices
+     */
+    void AppendDataColumns()
     {
-        typedef permutation_matrix<std::size_t> pmatrix;
-
-        // create a working copy of the input
-        T A(input);
-
-        // create a permutation matrix for the LU-factorization
-        pmatrix pm(A.size1());
-
-        // perform LU-factorization
-        int res = lu_factorize(A, pm);
-        if (res != 0)
-            return false;
-
-        // create identity matrix of "inverse"
-        inverse.assign(identity_matrix<double> (A.size1()));
+        // Create two auxilar observation matrices to reshape the existent ones
+        const std::size_t n_old_cols = mpObsMatrixV->size2();
+        MatrixPointerType p_new_V = Kratos::make_shared<MatrixType>(mProblemSize, n_old_cols + 1);
+        MatrixPointerType p_new_W = Kratos::make_shared<MatrixType>(mProblemSize, n_old_cols + 1);
+
+        // Recover the previous iterations information
+        for (unsigned int i = 0; i < mProblemSize; i++){
+            for (unsigned int j = 0; j < n_old_cols; j++){
+                (*p_new_V)(i,j) = (*mpObsMatrixV)(i,j);
+                (*p_new_W)(i,j) = (*mpObsMatrixW)(i,j);
+            }
+        }
 
-        // backsubstitute to get the inverse
-        lu_substitute(A, pm, inverse);
+        // Fill the attached column with the current iteration information
+        for (unsigned int i = 0; i < mProblemSize; i++){
+            (*p_new_V)(i, n_old_cols) = (*mpResidualVector_1)(i) - (*mpResidualVector_0)(i);
+            (*p_new_W)(i, n_old_cols) = (*mpIterationValue_1)(i) - (*mpIterationValue_0)(i);
+        }
 
-        return true;
+        std::swap(mpObsMatrixV,p_new_V);
+        std::swap(mpObsMatrixW,p_new_W);
     }
 
-    ///@}
-
-    ///@name Protected  Access
-    ///@{
-    ///@}
-
-    ///@name Protected Inquiry
-    ///@{
-    ///@}
+    /**
+     * @brief Drop the oldest data columns to append the new data
+     * This function drops the oldest information to append the new data to both observation matrices
+     */
+    void DropAndAppendDataColumns()
+    {
+        // Observation matrices size is equal to the interface DOFs number. Oldest columns are to be dropped.
+        MatrixPointerType p_new_V = Kratos::make_shared<MatrixType>(mProblemSize, mProblemSize);
```

Comment:
i am accepting but ... here you are copying the entire matrix at every iteration... this is a lot!!

Reply:
Is the unique way I found to do that. Indeed I don't think this is a problem, you only `DropAndAppendDataColumns`when your number of iterations is equal to your problem size. This means that the number of DOFs of your FSI interface is equal to the FSI non-linear iterations. Such situation only appears in extremely small toy problems (hopefully...)